runtime/compiler/control/CompilationThread.cpp

```diff
@@ -11141,31 +11143,38 @@ TR::CompilationInfoPerThreadBase::processException(
    catch (const JITServer::StreamFailure &e)
       {
       if (TR::Options::getVerboseOption(TR_VerboseJITServer))
-         TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer, "JITServer StreamFailure: %s", e.what());
+         TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer, "compThreadID=%d JITServer StreamFailure: %s", getCompThreadId(), e.what());
       _methodBeingCompiled->_compErrCode = compilationStreamFailure;
       }
    catch (const JITServer::StreamInterrupted &e)
       {
       if (TR::Options::getVerboseOption(TR_VerboseJITServer))
-         TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer, "JITServer StreamInterrupted: %s", e.what());
+         TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer, "compThreadID=%d JITServer StreamInterrupted: %s", getCompThreadId(), e.what());
       _methodBeingCompiled->_compErrCode = compilationStreamInterrupted;
       }
    catch (const JITServer::StreamVersionIncompatible &e)
       {
       if (TR::Options::getVerboseOption(TR_VerboseJITServer))
-         TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer, "JITServer StreamVersionIncompatible: %s", e.what());
+         TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer, "compThreadID=%d JITServer StreamVersionIncompatible: %s", getCompThreadId(), e.what());
       _methodBeingCompiled->_compErrCode = compilationStreamVersionIncompatible;
       }
    catch (const JITServer::StreamMessageTypeMismatch &e)
       {
       if (TR::Options::getVerboseOption(TR_VerboseJITServer))
-         TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer, "JITServer StreamMessageTypeMismatch: %s", e.what());
+         TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer, "compThreadID=%d JITServer StreamMessageTypeMismatch: %s", getCompThreadId(), e.what());
       _methodBeingCompiled->_compErrCode = compilationStreamMessageTypeMismatch;
       }
    catch (const JITServer::ServerCompilationFailure &e)
       {
-      // no need to set error code here because error code is set
+      // no need to set error code at JITClient because error code is set
       // in remoteCompile when the compilation failed
+      if (_methodBeingCompiled->isOutOfProcessCompReq())
+         {
+         if (TR::Options::getVerboseOption(TR_VerboseJITServer))
+            TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer, "compThreadID=%d JITServer ServerCompilationFailure: %s", getCompThreadId(), e.what());
+
+         _methodBeingCompiled->_compErrCode = compilationFailure;
```

Comment:
This error code is too generic. I am afraid that the client will not try to retry the compilation when it sees this code.

Reply:
When the exception `ServerCompilationFailure` happens, it could be because this change that throws the exception because of the unloaded classes, or it could be caused by other reasons. It's hard to decide on a specific error code to cover all cases when `ServerCompilationFailure` happens. 

I'm wondering if I should create a new exception instead of piggy-backing on `ServerCompilationFailure`. That would require more changes on the error handling path.

Reply:
Changed the exception to `TR::CompilationInterrupt`. The existing path will set `compErrCode`='compilationInterrupted`.